### DISCUSSION TITLE
Better waves by use of grayscale Displacement Maps

### DISCUSSION THREAD

**Discussion by FutureFireplace** [playcanvas/engine#4059] [Ideas]
Hello - if one were to use this method; https://www.youtube.com/watch?v=2AQLMZwQpDo, it would be possible to make more realistic ocean waves without use of vertex animation or actual heavy 3d series of objects etc. //br

----
As with Subsurface scattering there are still many areas where PlayCanvas can improve visually. I have used and will use a vertex shader for waves, but the solution is mediocre, as is has a very repetitive expression. With a grayscale solution, it would be possible to put refraction upon the model (by scripting and several layers for the specific involved/different models)

**Comment by yaustar** [comment]
Please use the forums for questions line this as it's not engine related. It's basically this under the hood https://developer.playcanvas.com/en/tutorials/terrain-generation/ but via a vertex shader
